src/main/java/io/numaproj/pulsar/producer/PulsarConfig.java

```
@@ -31,8 +37,24 @@ public PulsarClient pulsarClient(PulsarClientProperties pulsarClientProperties)
     @Bean
     public Producer<byte[]> pulsarProducer(PulsarClient pulsarClient, PulsarProducerProperties pulsarProducerProperties)
             throws Exception {
+        String podName = System.getenv("NUMAFLOW_POD");
+
+        if (podName == null || podName.isBlank()) {
+            podName = "pod-" + UUID.randomUUID();
+            log.warn("NUMAFLOW_POD environment variable not found. Generating fallback Producer name: {}", podName);
+        }
```

Is the case where a podName is not found a case we need to worry about? If so, should we throw an error or is this approach ok?

---

The case of NUMAFLOW_POD envvar not being found should never happen but I like the idea of checking it and fallback to a random name. LGTM!